### PR TITLE
Return error code for job start failures even with --detach

### LINKED ISSUE #1046: Return code is 0 when job didn't start
When `neuro run` has the `--detach` parameter it always returns zero exit code, even in situations when job was unable to run (e.g., when Cluster ScaleUp failed).

I think `neuro run` should return a non-zero exit code in situations when it knows job has failed.

Example:
```
# neuro run --name jupyter --detach …
Job ID: job-33c220ec-7c82-427e-939a-940490be59ac Status: pending
Name: jupyter
Http URL: https://jupyter--spbrspo.jobs-dev.neu.ro
…
Status: pending Initializing
Status: pending Scheduling
Status: failed ClusterScaleUpFailed (Failed to scale up the cluster to get more resources)
# echo $?
0
```

### PR DESCRIPTION
**Pull request by anayden**
Closes #1046

### CODE REVIEW

File: neuromation/cli/job.py [hunk 1]
```diff
@@ -919,8 +921,16 @@ def format_fail(job: str, reason: Exception) -> str:
...
         await _print_logs(root, job.id)
-        res = await root.client.jobs.status(job.id)
-        sys.exit(res.history.exit_code)
+        job = await root.client.jobs.status(job.id)
+        exit_code = job.history.exit_code
+    else:
+        # Even if we detached, but the job has failed to start
+        # (most common reason - no resources), the command fails
+        if job.status == JobStatus.FAILED:
+            exit_code = 1
```
Comment (asvetlov) [inline neuromation/cli/job.py]: I'm curious why you use `job.history.exit_code` for an attached job but `1` for detached one?

Comment (anayden) [reply neuromation/cli/job.py]: When detached job fails, it means it didn't even start. Hence there's no exit code. `job.history.exit_code` for such job is `None`

Comment (asvetlov) [reply neuromation/cli/job.py]: Got you.

Looks like docker uses `126` exit code for the case: https://docs.docker.com/engine/reference/run/#/exit-status
It is compliant to chroot exit codes table.
Maybe we can return `126` as well?

Comment (asvetlov) [reply neuromation/cli/job.py]: Or `125`, I have no strong opinion here.

Comment (anayden) [reply neuromation/cli/job.py]: I like 125, will update now.

**Comment by anayden** [comment]
A run of python 3.6 tests was successful. The tests are very flaky, but it's proven to work :) Merging.

**Pull request merged**